### ISSUE TITLE
LocalDateTime arithmetic

### ISSUE THREAD

**Issue by hfhbd** [Kotlin/kotlinx-datetime#66]
Currently, the ReadMe does not cover `LocalDateTime` arithmetic. Either make this arithmetic clear by adding this to the `Instant` arithmetic, with needs a timeZone.

```kotlin
val localDateTime: LocalDateTime = ...
val timeZone: TimeZone = ...
val futureLocalDateTime = (localDateTime.toInstant(timeZone) + 7.days).toLocalDateTime(timeZone)
```

Or add `LocalDateTime` arithmetic without a `TimeZone`, which can result into an overflow, like the `java.time` API.
```kotlin
val localDateTime: LocalDateTime = ...
val futureLocalDateTime = localDateTime + 7.days
```
```java
// java.time overflow arithmetic
public LocalDateTime plusHours(long hours) {
    return plusWithOverflow(date, hours, 0, 0, 0, 1);
}
```
I can submit a PR with `LocalDateTime` arithmetic, if wanted.

**Comment by dkhalanskyjb** [comment]
The problem with `LocalDateTime` arithmetic is that it is not well-defined. What does it mean to add two hours to a `LocalDateTime`? There could be two possible interpretations:
* Change the hour, adding `2` to it, possibly getting past midnight and starting a new day.
* Find out the time that the clock will show two hours after it has shown the source `LocalDateTime`.

Somewhat surprisingly, these two approaches give different results. The first one is easier to implement, but it is also often wrong. The reason for that is daylight saving time transitions: if clocks were moved an hour back "during" the addition, then adding two hours to the current time actually represents waiting for three hours.

It's easy to forget to handle daylight saving time, and the resulting errors are very annoying, as they won't show up in testing and will most likely cause problems in production. As `LocalDateTime` arithmetic that ignores DST is of limited use and encourages errors, we don't provide it.

> `val futureLocalDateTime = (localDateTime.toInstant(timeZone) + 7.days).toLocalDateTime(timeZone)`

This may be either correct or not correct, depending on what you wish to do here, and I'd guess that it's incorrect. In fact, it says to return the wall clock readings after `7 * 24` hours. If `localDateTime.hour == 14`, `futureLocalDateTime.hour` may be `13`, `14`, `15`, or, for fancy time zones, even something else. If you want to advance seven days without affecting time-of-day, a solution would be
```kotlin
val futureLocalDateTime = localDateTime.toInstant(timeZone)
    .plus(7, DateTimeUnit.DAY, timeZone)
    .toLocalDateTime(timeZone)
```
or
```kotlin
val futureLocalDateTime =
    localDateTime.date.plus(DatePeriod(days = 7))
        .atTime(localDateTime.hour, localDateTime.minute, localDateTime.second, localDateTime.nanosecond)
```
Though, yeah, the latter does look like something that should be easier to do. If we implement https://github.com/Kotlin/kotlinx-datetime/issues/57, this can become
```kotlin
// doesn't actually work currently!
val futureLocalDateTime =
    localDateTime.date.plus(DatePeriod(days = 7)).atTime(localDateTime.time)
```
Or maybe we could add `LocalDateTime.plus(DatePeriod)`.

The bottom line: you raised a good point, adding some date-only period to a `LocalDateTime` is more cumbersome than it has to be. Adding time-based periods to a `LocalDateTime` is intentionally cumbersome, as this is rarely what one really wants.
